// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package privacy_sandbox.server_common;

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

enum Severity {
  SEVERITY_UNKNOWN = 0;
  SEVERITY_INFO = 1;
  SEVERITY_WARNING = 2;
  SEVERITY_ERROR = 3;
}

message LogRequest {
  string input = 1;
  Severity severity = 2;
  int32 processing_delay_ms = 3;
}

message LogResponse {}

service LoggingService {
  rpc Log(LogRequest) returns (LogResponse) {}
}
